app/src/lib/stores/app-store.ts

```diff
@@ -444,34 +448,32 @@ export class AppStore extends TypedBaseStore<IAppState> {
   }
 
   private recordTutorialStepCompleted(step: TutorialStep): void {
-    switch (step) {
-      case TutorialStep.PickEditor:
-        this.statsStore.recordTutorialRepoCreated()
-        break
-      case TutorialStep.CreateBranch:
-        this.statsStore.recordTutorialEditorInstalled()
-        break
-      case TutorialStep.EditFile:
-        this.statsStore.recordTutorialBranchCreated()
-        break
-      case TutorialStep.MakeCommit:
-        this.statsStore.recordTutorialFileEdited()
-        break
-      case TutorialStep.PushBranch:
-        this.statsStore.recordTutorialCommitCreated()
-        break
-      case TutorialStep.OpenPullRequest:
-        this.statsStore.recordTutorialBranchPushed()
-        break
-      case TutorialStep.AllDone:
-        this.statsStore.recordTutorialPrCreated()
-        this.statsStore.recordTutorialCompleted()
-        break
-      case TutorialStep.NotApplicable:
-      case TutorialStep.Paused:
-        break
-      default:
-        assertNever(step, 'Unaccounted for step type')
+    // if it's not a valid step, we don't need to record anything
+    if (!isValidTutorialStep(step)) {
+      return
+    }
+
+    const stepInd = orderedTutorialSteps.indexOf(step)
+    this.statsStore.recordHighestTutorialStepCompleted(stepInd)
+
+    if (stepInd > orderedTutorialSteps.indexOf(TutorialStep.PickEditor)) {
```

Comment:
Non-blocking suggestion:

We could store off all of these indices as variables to reduce having to compute them every time we record steps as completed. For example:

```suggestion
    if (stepInd > PickEditorStepIndex) {
```